### PR TITLE
add documentation for spaceport

### PR DESCRIPTION
**Pull request by garypen**
Minimal documentation to support spaceport configuration.

### CODE REVIEW

**Review by o0Ignition0o — APPROVED** [review]

**Review by Geal — APPROVED** [review]

File: docs/source/configuration.mdx [hunk 1]
```diff
@@ -261,3 +248,58 @@ opentelemetry:
...
+  key: service:<redacted>
+```
+
+You must configure the graph section. If you specify a graph section, then both the reference and the key must be specified. The reference must refer to a valid graph/version combination against which statistics are reported. The Key must be a valid API key which can be used for reporting statistics.
+For more details on this process, refer to the [usage reporting documentation](https://www.apollographql.com/docs/studio/metrics/usage-reporting/)
+The key fulfills the same function as the APOLLO_KEY environment variable.
+The reference fulfills the same function as the APOLLO_GRAPH_REF environment variable.
+NB: These environment variables are *not* currently recognised by the router. This is being tracked by [issue](https://github.com/apollographql/router/issues/425)
+
+### Spaceport Configuration
```
Comment (abernix) [inline docs/source/configuration.mdx:282]:
```suggestion
### Spaceport configuration
```

File: docs/source/configuration.mdx [hunk 2]
```diff
@@ -261,3 +248,58 @@ opentelemetry:
...
+spaceport:
+  # Should we spawn a Spaceport or talk to an existing external Spaceport?
+  external: false
+  # What's the address of the Spaceport?
+  collector: https://127.0.0.1:50051
+  # Where should we listen IF we have spawned an internal Spaceport?
+  listener: 0.0.0.0:50051
+```
+
+#### Internal Spaceport
```
Comment (abernix) [inline docs/source/configuration.mdx:297]:
```suggestion
#### Internal spaceport
```

File: docs/source/configuration.mdx [hunk 3]
```diff
@@ -261,3 +248,58 @@ opentelemetry:
...
+
+This is the default and is most appropriate for evaluations, demos, development support and generally getting started. This requires no configuration by default and will be automatically started if the graph configuration is present.
+In this mode it is possible to further configure the address at which the internal spaceport is listening and the address of the spaceport to use. This should only be necessary if there is a clash on the default port, for instance if running multiple routers on the same system or clashing with another application.
+
+#### External Spaceport
```
Comment (abernix) [inline docs/source/configuration.mdx:302]:
```suggestion
#### External spaceport
```

File: docs/source/configuration.mdx [hunk 4]
```diff
@@ -261,3 +248,58 @@ opentelemetry:
                   - value1
                   - value2
 ```
+
+## Spaceport
+
+Spaceport reports usage data to the Apollo ingress. You can see how it is configured in the examples above.
+
+NB: This component is still evolving and you should expect functionality to change and evolve here. We are actively seeking user feedback as we work to shape this component.
```
Comment (abernix) [inline docs/source/configuration.mdx:256]:
```suggestion
> Note: This component is still evolving and you should expect functionality to change and evolve here. We are actively seeking user feedback as we work to shape this component.
```

File: docs/source/configuration.mdx [hunk 5]
```diff
@@ -261,3 +248,58 @@ opentelemetry:
...
+# processing, since there is no point collecting statistics unless we have
+# somewhere to send them.
+graph:
+  # Which graph should our Spaceport be sending data about?
+  reference: Benchmarking@current
+  # Key used to send data
+  key: service:<redacted>
+```
+
+You must configure the graph section. If you specify a graph section, then both the reference and the key must be specified. The reference must refer to a valid graph/version combination against which statistics are reported. The Key must be a valid API key which can be used for reporting statistics.
```
Comment (abernix) [inline docs/source/configuration.mdx:276]:
```suggestion
You must configure the `graph` section. If you specify a `graph` section, then both the `reference` and the `key` must be specified. The `reference` must refer to a valid graph/variant combination (i.e., the "graph ref") against which statistics are reported. The `key` must be a valid Apollo Studio API key which can be used for reporting statistics (a [Studio Graph _Contributor_ token](https://www.apollographql.com/docs/studio/org/members/#organization-wide-member-roles) is appropriate).
```

File: docs/source/configuration.mdx [hunk 6]
```diff
@@ -261,3 +248,58 @@ opentelemetry:
...
+For more details on this process, refer to the [usage reporting documentation](https://www.apollographql.com/docs/studio/metrics/usage-reporting/)
+The key fulfills the same function as the APOLLO_KEY environment variable.
```
Comment (abernix) [inline docs/source/configuration.mdx:278]:
```suggestion
The key fulfills the same function as the `APOLLO_KEY` environment variable.
```

File: docs/source/configuration.mdx [hunk 7]
```diff
@@ -261,3 +248,58 @@ opentelemetry:
...
+The reference fulfills the same function as the APOLLO_GRAPH_REF environment variable.
```
Comment (abernix) [inline docs/source/configuration.mdx:279]:
```suggestion
The `reference` fulfills the same function as the `APOLLO_GRAPH_REF` environment variable.
```

File: docs/source/configuration.mdx [hunk 8]
```diff
@@ -261,3 +248,58 @@ opentelemetry:
...
+NB: These environment variables are *not* currently recognised by the router. This is being tracked by [issue](https://github.com/apollographql/router/issues/425)
```
Comment (abernix) [inline docs/source/configuration.mdx:280]:
```suggestion
> Note: These environment variables are *not* currently recognised by the router. This is being tracked by [#425](https://github.com/apollographql/router/issues/425)
```

**Comment by abernix** [comment]
Meh, well, my suggested changes can't be auto-clicked in now, but it was mostly formatting.

**Pull request merged**